Mathlib/Algebra/Category/Grp/Ulift.lean

```diff
@@ -0,0 +1,136 @@
+
+/--
+The universe lift functor on `Grp.{u}` creates `u`-small limits.
+-/
+@[to_additive
+  "The universe lift functor on `AddGrp.{u}` creates `u`-small limits."]
+noncomputable instance : CreatesLimitsOfSize.{w, u} uliftFunctor.{v, u} where
+  CreatesLimitsOfShape := { CreatesLimit := fun {_} ↦ createsLimitOfFullyFaithfulOfPreserves }
+
+end Grp
+
+namespace CommGrp
+
+/-- The universe lift functor for commutative groups is fully faithful.
+-/
+@[to_additive
+  "The universe lift functor for commutative additive groups is fully faithful."]
+def uliftFunctorFullyFaithful : uliftFunctor.{u, v}.FullyFaithful where
+  preimage f := Grp.ofHom (MulEquiv.ulift.toMonoidHom.comp
+    (f.comp MulEquiv.ulift.symm.toMonoidHom))
+  map_preimage _ := rfl
+  preimage_map _ := rfl
+
+-- The universe lift functor for commutative groups is faithful. -/
+@[to_additive
+  "The universe lift functor for commutative additive groups is faithful."]
+instance : uliftFunctor.{u, v}.Faithful := Functor.FullyFaithful.faithful uliftFunctorFullyFaithful
+
+-- The universe lift functor for commutative groups is full. -/
+@[to_additive
+  "The universe lift functor for commutative additive groups is full."]
+instance : uliftFunctor.{u, v}.Full := Functor.FullyFaithful.full uliftFunctorFullyFaithful
+
+@[to_additive]
+noncomputable instance uliftFunctor_preservesLimit {J : Type w} [Category.{w'} J]
+    (K : J ⥤ CommGrp.{u}) : PreservesLimit K uliftFunctor.{v, u} where
+  preserves lc := ⟨isLimitOfReflects (forget CommGrp.{max u v})
+    (isLimitOfPreserves CategoryTheory.uliftFunctor (isLimitOfPreserves (forget CommGrp) lc))⟩
+
+@[to_additive]
+noncomputable instance uliftFunctor_preservesLimitsOfShape {J : Type w} [Category.{w'} J] :
+    PreservesLimitsOfShape J uliftFunctor.{v, u} where preservesLimit := inferInstance
+
+/--
+The universe lift for commutative groups preserves limits of arbitrary size.
+-/
+@[to_additive
+  "The universe lift functor for commutative additive groups preserves limits of arbitrary size."]
+noncomputable instance uliftFunctor_preservesLimitsOfSize :
+    PreservesLimitsOfSize.{w', w} uliftFunctor.{v, u} where preservesLimitsOfShape := inferInstance
+
+/--
+The universe lift functor on `CommGrp.{u}` creates `u`-small limits.
+-/
+@[to_additive
+  "The universe lift functor on `AddCommGrp.{u}` creates `u`-small limits."]
+noncomputable instance : CreatesLimitsOfSize.{w, u} uliftFunctor.{v, u} where
+  CreatesLimitsOfShape := { CreatesLimit := fun {_} ↦ createsLimitOfFullyFaithfulOfPreserves }
+
+end CommGrp
+
+namespace AddCommGroup
+
+/-- The universe lift for commutative additive groups preserves zero morphisms.
+-/
+instance uliftFunctor_preservesZeroMorphisms :
+    AddCommGrp.uliftFunctor.{u,v}.PreservesZeroMorphisms where map_zero _ _ := rfl
+
+instance uliftFunctor_additive :
+    AddCommGrp.uliftFunctor.{u,v}.Additive where map_add := rfl
```

Comment:
```suggestion
instance uliftFunctor_additive :
    AddCommGrp.uliftFunctor.{u,v}.Additive where
```
If a functor is additive, mathlib automatically knows it preserves zero morphisms.